app/bundles/LeadBundle/EventListener/ReportSubscriber.php

```diff
@@ -236,6 +236,10 @@ public function onReportBuilder(ReportBuilderEvent $event)
             if ($event->checkContext(['leads', 'contact.frequencyrules'])) {
                 $this->injectFrequencyReportData($event, $columns);
             }
+
+            if ($event->checkContext(['leads', 'segment.membership'])) {
+                $this->injectSegmentReportData($event, $columns);
```

Comment:
We want the list of segments as a dropdown in the Filters so I think maybe `$filters` https://github.com/mautic/mautic/pull/5157/files#diff-02107509b7f696d7335178eb82e3b15bR187 needs to be sent into this method as well.